boards/kernel_layout.ld

```diff
@@ -337,6 +337,40 @@ SECTIONS
     } > ram
     _eappmem = ORIGIN(ram) + LENGTH(ram);
 
+
+    /* Place attributes at the end of the kernel ROM region. These will be used
+     * for host-side tools to learn about the installed kernel.
+     *
+     * Attributes are stored as TLVs, but going backwards in flash.
+     *
+     *                           2 byte   2 byte   4 byte
+     * -----------+------------+--------+--------+------------+
+     *            | Value...   | Type   | Length | TOCK       | Start of `prog`
+     * -----------+------------+--------+--------+------------+
+     * <-TLVs...-> <-----TLV--------------------> <-Sentinel->
+     *
+     * `Length` specifies the size of the value (i.e. not include the header).
+     */
+    .attributes :
+    {
+        . = ABSOLUTE(ORIGIN(rom) + LENGTH(rom) - 16);
+
+        /* TLV: App Memory
+         * This indicates the start address and size of RAM available for
+         * userspace apps.
+         */
+        LONG(_sappmem) /* Address of start of app memory. */
+        LONG(_eappmem - _sappmem) /* Length of app memory. */
+        SHORT(0x0101) /* Type = App RAM Region = 0x0101 */
+        SHORT(8)      /* Length = 8 bytes */
+
+        BYTE(84) /* T */
+        BYTE(79) /* O */
+        BYTE(67) /* C */
+        BYTE(75) /* K */
+    }  > rom
+
```

Comment:
I have:
```
ASSERT((_etext-_stext) + (_erelocate-_srelocate) + (_eattributes-_sattributes) < LENGTH(rom),
"Text plus relocations plus attributes exceeds the available ROM space.");
ASSERT(_sattributes > _erelocate, "Kernel attributes overlaps relocations.");
```

I'm not sure it's possible to add more, since if they are the same then whatever is first will trigger and the rest are superfluous. And the linker won't let you overflow rom.

Reply:
"And the linker won't let you overflow rom." That's great. I didn't think about that. So if we change things and forget to update the 16 it'll already fail at build time.

Reply:
Ooh, you're right. Although it could have been `_etext + (_erelocate-_srelocate)` then.

Reply:
The linker will already check that it doesn't go beyond ROM or overlap other sections.

I added a check that .attributes is actually at the end of ROM.

Reply:
Yeah, these asserts are good. Thanks for thinking through those.